features/flows/classification/engine/src/main/java/org/opennms/netmgt/flows/classification/DefaultClassificationEngine.java

```diff
@@ -31,27 +31,26 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 
 import org.opennms.netmgt.flows.classification.classifier.Classifier;
 import org.opennms.netmgt.flows.classification.classifier.CombinedClassifier;
-import org.opennms.netmgt.flows.classification.matcher.IpMatcher;
-import org.opennms.netmgt.flows.classification.matcher.Matcher;
-import org.opennms.netmgt.flows.classification.matcher.PortMatcher;
-import org.opennms.netmgt.flows.classification.matcher.ProtocolMatcher;
 import org.opennms.netmgt.flows.classification.persistence.api.Rule;
 import org.opennms.netmgt.flows.classification.provider.ClassificationRuleProvider;
 import org.opennms.netmgt.flows.classification.provider.StaticClassificationRuleProvider;
+import org.opennms.netmgt.flows.classification.value.PortValue;
 import org.slf4j.LoggerFactory;
 
 import com.google.common.base.Strings;
 
 public class DefaultClassificationEngine implements ClassificationEngine {
 
     private final ClassificationRuleProvider ruleProvider;
-    private final List<Classifier> classifierList = new ArrayList<>();
+    private final Map<Integer, List<Classifier>> classifierPortMap = new HashMap<>();
```

Comment:
Since our keys are integers and there are a fixed number of ports (2^16), we could also use an ArrayList here - this could help reduce the lookup time.